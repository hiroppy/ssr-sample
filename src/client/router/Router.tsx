import React from 'react';
import { Route, Switch } from 'react-router-dom';
import {
  LoadableTopPage,
  LoadableSagaPage,
  LoadableApolloPage,
  LoadableNotFoundPage
} from './routes';
import { App } from '../components/App';

export const Router = () => (
  <App>
    <Switch>
      <Route exact path="/">
        <LoadableTopPage />
      </Route>
      <Route path="/saga">
        <LoadableSagaPage />
      </Route>
      <Route path="/apollo">
        <LoadableApolloPage />
      </Route>
      <Route>
        <LoadableNotFoundPage />
      </Route>
    </Switch>
  </App>
);
