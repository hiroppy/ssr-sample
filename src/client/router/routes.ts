import loadable from '@loadable/component';

export const LoadableTopPage = loadable(() =>
  import(/* webpackPrefetch: true */ '../components/pages/Top')
);

export const LoadableSagaPage = loadable(() =>
  import(/* webpackPrefetch: true */ '../components/pages/Saga')
);

export const LoadableApolloPage = loadable(() =>
  import(/* webpackPrefetch: true */ '../components/pages/Apollo')
);

export const LoadableNotFoundPage = loadable(() => import('../components/pages/NotFound'));
