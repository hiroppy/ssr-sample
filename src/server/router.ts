import * as express from 'express';
import { ApolloServer } from 'apollo-server-express';
import * as health from './controllers/health';
import * as renderer from './controllers/renderer';
import { resolvers, schema } from '../graphql/schema';
import { endpoint } from '../graphql/constants';

const apollo = new ApolloServer({
  debug: process.env.NODE_ENV !== 'production',
  schema,
  resolvers,
  playground:
    process.env.NODE_ENV !== 'production'
      ? {
          endpoint
        }
      : false,
  // logging
  // wait for https://github.com/apollographql/apollo-server/pull/1748
  formatError: (err) => {
    // console.log(JSON.stringify(err, null, 2));
    return err;
  },
  formatResponse: (res: { data: any }) => {
    // console.log(JSON.stringify(res.data, null, 2));
    return res;
  }
});

export function router(app: express.Application) {
  apollo.applyMiddleware({ app, path: endpoint });

  app.use('/favicon.ico', (req, res) => res.status(200).send());
  app.use('/public', express.static('dist/client'));
  app.get('/api/health', health.get);
  app.get('*', renderer.get);
}
