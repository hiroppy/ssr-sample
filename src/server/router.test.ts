import express from 'express';
import { router } from './router';

test('should register url paths', () => {
  const app = express();

  router(app);

  const res = app._router.stack.map((r: { name: string; regexp: RegExp }) => {
    return {
      name: r.name,
      path: r.regexp
    };
  });

  expect(res).toMatchSnapshot();
});
