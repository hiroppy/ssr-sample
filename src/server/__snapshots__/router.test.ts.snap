// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should register url paths 1`] = `
Array [
  Object {
    "name": "query",
    "path": /\\^\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
  },
  Object {
    "name": "expressInit",
    "path": /\\^\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
  },
  Object {
    "name": "<anonymous>",
    "path": /\\^\\\\/graphql\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
  },
  Object {
    "name": "<anonymous>",
    "path": /\\^\\\\/\\\\\\.well-known\\\\/apollo\\\\/server-health\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
  },
  Object {
    "name": "corsMiddleware",
    "path": /\\^\\\\/graphql\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
  },
  Object {
    "name": "jsonParser",
    "path": /\\^\\\\/graphql\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
  },
  Object {
    "name": "<anonymous>",
    "path": /\\^\\\\/graphql\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
  },
  Object {
    "name": "<anonymous>",
    "path": /\\^\\\\/graphql\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
  },
  Object {
    "name": "<anonymous>",
    "path": /\\^\\\\/favicon\\\\\\.ico\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
  },
  Object {
    "name": "serveStatic",
    "path": /\\^\\\\/public\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
  },
  Object {
    "name": "bound dispatch",
    "path": /\\^\\\\/api\\\\/health\\\\/\\?\\$/i,
  },
  Object {
    "name": "bound dispatch",
    "path": /\\^\\(\\.\\*\\)\\\\/\\?\\$/i,
  },
]
`;
